cpp/oneapi/dal/test/engine/dataframe_common.cpp

```diff
@@ -273,6 +297,19 @@ dataframe_builder_impl::dataframe_builder_impl(std::int64_t row_count, std::int6
     program_.add<dataframe_builder_action_allocate>(row_count, column_count);
 }
 
+dataframe_builder_impl::dataframe_builder_impl(std::string dataset) {
+    const char* dataset_root_ptr = std::getenv("DATASETSROOT");
+    if (dataset_root_ptr == nullptr) {
+        throw invalid_argument{ "DATASETSROOT environment variable is unset" };
+    }
+    const std::string datasets_root(dataset_root_ptr);
+
+    const auto data_table = dal::read<dal::table>(dal::csv::data_source{ datasets_root + dataset });
+    program_.add<dataframe_builder_action_set>(dal::row_accessor<const float>{ data_table }.pull(),
+                                               data_table.get_row_count(),
+                                               data_table.get_column_count());
+}
```

Comment:
Do not recommend splitting logic and responsibilities between builder and action.

1. According to current design and implementation of the methods, builder has straightforward responsibilities: create action and add to the program. So move dataset reading logic to separate action.
2. This action should include dataset name as part of opcode, see `get_opcode()` method to support proper caching